### ISSUE TITLE
list_dir() hangs if folder is empty

### ISSUE THREAD

**Issue by mikl** [i22-digitalagentur/sftp_client#24]
Hi there,

I’m experiencing something weird. I can connect to SFTP servers just fine, all the operations I’ve tried work, but when I do a list_dir command, it hangs forever if the folder contains no files.

For example this:

```elixir
    {:ok, conn} = SFTPClient.connect(get_in(state, [:config, :server]))
    {:ok, stat} = SFTPClient.file_info(conn, "in/images/products")
    IO.puts("Stat result #{inspect(stat)}")
    {:ok, list} = SFTPClient.list_dir(conn, "in/images/products")
    IO.inspect(list)
    SFTPClient.disconnect(conn)
```

If the `"in/images/products"` folder contains files, everything is hunky-dory, I get a list back.

However, if `"in/images/products"` has no files (either being completely empty, or having just subfolders), list_dir does not return. The following lines are not executed, I’ve tried waiting at least 10 minutes, much longer than my SFTP timeout, so something’s weird.

Let me know if there’s anything I can do to help reproduce this problem.

**Comment by mikl** [comment]
A further clue, when I set `operation_timeout: 5000` explicitly in my server config, I do get a `{:error, %SFTPClient.OperationError{reason: :timeout}}` response. 

The docs at https://hexdocs.pm/sftp_client/1.4.6/SFTPClient.Operations.Connect.html#connect/1 state that `5000` is the default value, but that does not seem to apply here?

I realise this might be problem in the Erlang parts, my Erlang and Elixir versions are:

```
elixir --version
Erlang/OTP 23 [erts-11.2.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]
Elixir 1.11.4 (compiled with Erlang/OTP 23)
```

Both installed with ASDF on macOS 11.3.1 (20E241) on Intel 64bit.

And my installed version of sftp_client is the latest (1.4.6).

**Comment by tlux** [comment]
Thanks for the bug report!

Not sure what is happening here as sftp_client is just forwarding to the respective Erlang calls. 
It may take me a couple of days to start digging into this.

**Comment by mikl** [comment]
On further investigation, I have determined that this particular SFTP server appears to be cursed. 

Even using the Erlang ssh_sftp directly does not work:

```elixir
:ssh.start()

{:ok, channel_pid, _connection} =
  :ssh_sftp.start_channel(
    'lobster-test.some.net',
    2222,
    user: 'userr',
    password: 'ppassword'
  )

IO.puts("Successfully connected to SFTP server.")

{:ok, file_names} =
  :ssh_sftp.list_dir(
    channel_pid,
    '/home/some/folder/whatever',
    10_000
  )
```

Same timeout error. So this is not a bug in this library, sorry for the trouble.

TL;DR: Lobster _data’s SFTP server is weird.

**Comment by shaharz** [comment]
FWIW setting `sftp_vsn: 3` as suggested [here](https://elixirforum.com/t/no-response-from-ssh-sftp-list-dir-2-when-given-path-other-than-root/25130/5) did the trick in my case.
